OpenRA.Mods.Common/Lint/CheckLocomotorReferences.cs

```diff
@@ -0,0 +1,51 @@
+#region Copyright & License Information
+/*
+ * Copyright 2007-2018 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation, either version 3 of
+ * the License, or (at your option) any later version. For more
+ * information, see COPYING.
+ */
+#endregion
+
+using System;
+using System.Linq;
+using OpenRA.Mods.Common.Traits;
+using OpenRA.Traits;
+
+namespace OpenRA.Mods.Common.Lint
+{
+	public class CheckLocomotorReferences : ILintRulesPass
+	{
+		public void Run(Action<string> emitError, Action<string> emitWarning, Ruleset rules)
+		{
+			foreach (var actorInfo in rules.Actors)
+			{
+				foreach (var traitInfo in actorInfo.Value.TraitInfos<ITraitInfo>())
+				{
+					var fields = traitInfo.GetType().GetFields().Where(f => f.HasAttribute<LocomotorReferenceAttribute>());
+					foreach (var field in fields)
+					{
+						var locomotors = LintExts.GetFieldValues(traitInfo, field, emitError);
+						foreach (var locomotor in locomotors)
+						{
+							if (string.IsNullOrEmpty(locomotor))
+								continue;
+
+							CheckLocomotors(actorInfo.Value, emitError, rules, locomotor);
+						}
+					}
+				}
+			}
+		}
+
+		void CheckLocomotors(ActorInfo actorInfo, Action<string> emitError, Ruleset rules, string locomotor)
+		{
+			var worldActor = rules.Actors["world"];
+			var locomotorInfos = worldActor.TraitInfos<LocomotorInfo>();
```

Comment:
Minor: You could pull this out of all the foreach loops probably, as this isn't changing during the lint run?